packages/lesswrong/lib/collections/users/callbacks.js

```diff
@@ -139,3 +139,35 @@ async function subscribeOnSignup (user) {
   }
 }
 addCallback('users.new.async', subscribeOnSignup);
+
+async function handleSetShortformPost (newUser, oldUser) {
+  if (newUser.shortformFeedId !== oldUser.shortformFeedId)
+  {
+    const post = await Posts.findOne({_id: newUser.shortformFeedId});
+    if (!post)
+      throw new Error("Invalid post ID for shortform");
+    if (post.userId !== newUser._id)
+      throw new Error("Post can only be an author's short-form post if they are the post's author");
+    if (post.draft)
+      throw new Error("Draft post cannot be a user's short-form post");
+    if (post.deleted)
+      throw new Error("Deleted post cannot be a user's short-form post");
+    
+    // In theory, we should check here whether the user already had a short-form
```

Comment:
I expect we will have to add this functionality eventually, but it's fine for now